cypress/e2e/hub/collections.cy.ts

```diff
@@ -120,30 +178,55 @@ describe('Collections List- Line Item Kebab Menu', () => {
       cy.clickButton('Close');
     });
     cy.clickButton('Clear all filters');
-    cy.visit(`/collections?page=1&perPage=50&sort=name&keywords=${thisCollectionName}`);
+    cy.visit(`/collections?page=1&perPage=50&sort=name&keywords=${collection}`);
     cy.get('[data-cy="table-view"]').click();
     cy.contains('h2', 'No results found').should('be.visible');
     repository = 'published';
   });
 
   it.skip('user can copy a version to repository', () => {
     //skipping this test because the Copy to Repository option is disabled for admin user
-    cy.approveCollection(thisCollectionName, namespace, '1.0.0');
-    cy.collectionCopyVersionToRepositories(thisCollectionName);
+    cy.approveCollection(collection, namespace, '1.0.0');
+    cy.collectionCopyVersionToRepositories(collection);
     repository = 'community';
   });
 });
 
-describe.skip('Collections Details View', () => {
-  before(() => {
+describe('Collections Details View', () => {
+  let collection: string;
+  let namespace: string;
+
+  beforeEach(() => {
+    collection = 'hub_e2e_' + randomString(5).toLowerCase();
+    namespace = 'hub_e2e_namespace' + randomString(5).toLowerCase();
     cy.hubLogin();
+    cy.createNamespace(namespace);
+    cy.uploadCollection(collection, namespace);
+    cy.navigateTo('hub', Collections.url);
+  });
+
+  afterEach(() => {
+    cy.galaxykit('task wait all');
+    cy.deleteNamespace(namespace);
   });
 
   it.skip('user can upload a new version to an existing collection', () => {});
 
-  it.skip('user can delete entire collection from system', () => {});
+  it('user can delete entire collection from system', () => {
+    cy.approveCollection(collection, namespace, '1.0.0');
+    cy.get('[data-cy="table-view"]').click();
+    cy.clickTableRow(collection);
+    cy.wait(400);
```

Comment:
Could we just do an assertion? To avoid wait, the reason why is because cypress will retry if an assertion fails, but the wait is pure heuristics.


cy.get('StringThatShouldDisplayed')

Reply:
I put this in because the details screen reloads a few times before stabilizing, so if I put a string that should be displayed it will work but still not be able to find the next assertion until it finishes reloading- unless there is a way to say 'wait until page finishes loading'?